lib/cpp/src/thrift/server/TThreadedServer.cpp

```diff
@@ -77,44 +83,29 @@ TThreadedServer::TThreadedServer(const shared_ptr<TProcessor>& processor,
                                  const shared_ptr<TProtocolFactory>& inputProtocolFactory,
                                  const shared_ptr<TProtocolFactory>& outputProtocolFactory,
                                  const shared_ptr<ThreadFactory>& threadFactory)
-  : TServerFramework(processor,
-                     serverTransport,
-                     inputTransportFactory,
-                     outputTransportFactory,
-                     inputProtocolFactory,
-                     outputProtocolFactory),
-    threadFactory_(threadFactory) {
+  : TThreadPoolServer(processor,
+                      serverTransport,
+                      inputTransportFactory,
+                      outputTransportFactory,
+                      inputProtocolFactory,
+                      outputProtocolFactory,
+                      apache::thrift::concurrency::ThreadManager::newSimpleThreadManager(0, 0)) {
+    threadManager_->threadFactory(threadFactory);
+    threadManager_->start();
 }
 
 TThreadedServer::~TThreadedServer() {
 }
 
-void TThreadedServer::serve() {
-  TServerFramework::serve();
-
-  // Drain all clients - no more will arrive
-  try {
-    Synchronized s(clientsMonitor_);
-    while (getConcurrentClientCount() > 0) {
-      clientsMonitor_.wait();
-    }
-  } catch (TException& tx) {
-    string errStr = string("TThreadedServer: Exception joining workers: ") + tx.what();
-    GlobalOutput(errStr.c_str());
+void TThreadedServer::onClientConnected(const shared_ptr<TConnectedClient>& pClient) {
+  if (!threadManager_->idleWorkerCount())
```

Comment:
I would personally like to see how the threads would be reaped before we go down this path, but that's mostly because I am unfamiliar with the thread manager. I'll defer to you and the maintainers on whether to do this without the accompanying thread manager change.